server/src/test/java/org/elasticsearch/snapshots/SnapshotResiliencyTests.java

```diff
@@ -517,6 +518,61 @@ public void testConcurrentSnapshotCreateAndDeleteOther() {
         }
     }
 
+    public void testConcurrentSnapshotRestoreAndDeleteOther() {
+        setupTestCluster(randomFrom(1, 3, 5), randomIntBetween(2, 10));
+
+        String repoName = "repo";
+        String snapshotName = "snapshot";
+        final String index = "test";
+        final int shards = randomIntBetween(1, 10);
+
+        TestClusterNodes.TestClusterNode masterNode =
+            testClusterNodes.currentMaster(testClusterNodes.nodes.values().iterator().next().clusterService.state());
+
+        final StepListener<CreateSnapshotResponse> createSnapshotResponseStepListener = new StepListener<>();
+
+        continueOrDie(createRepoAndIndex(repoName, index, shards),
+            createIndexResponse -> client().admin().cluster().prepareCreateSnapshot(repoName, snapshotName)
```

Comment:
Should we also index some docs (mostly to generate more snapshot files) before and in-between snapshots, and then run a query?

Reply:
Good idea, right now I don't expect it to make a difference but good to have some checks on this :) I pushed e8fe17ecd03

Reply:
I don't expect it either but I prefer to have some shard files around :) Thanks